src/meta_memcache/base/base_cache_pool.py

```diff
@@ -206,6 +209,12 @@ def _conn_recv_response(
             return Success(flags=set([Flag.NOREPLY]))
         result = conn.get_response()
         if isinstance(result, Value):
+            if flags and Flag.RETURN_KEY in flags:
+                if result.token_flags.get(TokenFlag.KEY) != key.key.encode():
+                    _log.error(
```

Comment:
One random question, are we expecting to see these logs somewhere in production?

Reply:
it should not happen, no... But messing with gevent profilingI saw errors caused by values that were not supposed to be the right response for a key, like it was messing up responses. So having this will make sure a wrong value doesn't get into the application and is understand as something it shouldn't.